src/nupic/research/temporal_memory.py

```diff
@@ -406,19 +459,19 @@ def leastUsedCell(cells, connections, random):
 
 
   @staticmethod
-  def growSynapses(connections, initialPermanence, nDesiredNewSynapes,
-                   prevWinnerCells, random, segment):
+  def growSynapses(connections, random, segment, nDesiredNewSynapes,
+                   prevWinnerCells, initialPermanence):
```

Comment:
Okay. Typically I'd push back b/c changing non-keyworded arguments can cause issues for backwards compatibility, not to mention for "public" methods. However, tests pass and coherence between the Python and C++ versions is nice, so all good.